### ISSUE TITLE
False Positive: abstract attribute error when class isn't ABCMeta

### ISSUE THREAD

**Issue by jamesbraza** [python/mypy#11471]
**Bug Report**

mypy raises an error when one tries to instantiate a class that fails to override an `abc.abstractmethod`.

Note that Python only enforces `abc.abstractmethod` when the class's metaclass is `ABCMeta`.

I think mypy should only raise an error when the class:
- Fails to override an `abc.abstractmethod`
- And that class inherits has the metaclass `ABCMeta`

**To Reproduce**

```python
import abc

class Foo:  # Note: not an abc.ABC
    @abc.abstractmethod
    def method(self) -> None:
        pass

foo = Foo()  # error: Cannot instantiate abstract class "Foo" with abstract attribute "method"  [abstract]
```

**Expected Behavior**

No error message, as `Foo` doesn't inherit from `abc.ABC`, so it can be instantiated as is.

**Actual Behavior**

Here is the output of mypy:

```
8: error: Cannot instantiate abstract class "Foo" with abstract attribute "method"  [abstract]
```

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: n/a
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: 3.8.12
- Operating system and version: macOS Big Sur, version 11.5.1

**Comment by JelleZijlstra** [comment]
I'm not sure mypy is wrong here. The class is still conceptually abstract when it has an unimplemented method, so instantiating it is unsafe.

**Comment by sobolevn** [comment]
I agree. I use this technique when I can possibly have metaclass conflicts and I only care about type-checking.
So, I use `@abcstractmethod` with no `ABCMeta`.

And I think that it is great!

**Comment by AlexWaygood** [comment]
This is a deliberate feature so that somebody can make a class "statically abstract" without having to have a special metaclass at runtime.
